docs/pages/x/api/tree-view/rich-tree-view.json

```diff
@@ -94,14 +94,31 @@
     "import { RichTreeView } from '@mui/x-tree-view/RichTreeView';",
     "import { RichTreeView } from '@mui/x-tree-view';"
   ],
-  "classes": [
+  "slots": [
     {
-      "key": "root",
-      "className": "MuiRichTreeView-root",
-      "description": "Styles applied to the root element.",
-      "isGlobal": false
+      "name": "root",
+      "description": "Element rendered at the root.",
+      "default": "RichTreeViewRoot",
+      "class": "MuiRichTreeView-root"
+    },
+    {
+      "name": "item",
+      "description": "Custom component for the item.",
+      "default": "TreeItem.",
+      "class": null
+    },
+    {
+      "name": "collapseIcon",
+      "description": "The default icon used to collapse the node.",
+      "class": null
+    },
+    {
+      "name": "expandIcon",
+      "description": "The default icon used to expand the node.",
+      "class": null
     }
   ],
+  "classes": [],
```

Comment:
Is it correct that the `*TreeView` components lose the `classes`? 🤔 
It does seem that they support them. 🤷

Reply:
seems ok, since there was only one and we can now use the `root` slot as well to apply classes! 👍🏼

Reply:
No it's not normal...
And I'm very curious to see why exporting the slots resulted in not generating the doc of the classes...

@michelengelen what do you mean by "we can now use the root slot as well to apply classes!" ?

Reply:
OK I think I get how it works
We only list the classes that don't have a slot because otherwise they are handled by the slot

![image](https://github.com/mui/mui-x/assets/3309670/bdca8e89-5358-4911-81e5-bae18995a410)


And for those components, each class has a slot

Reply:
sry, I probably approved a bit to early.

Reply:
Thanks for clarifying. 👍 
That's something new, I did not notice this merged behavior. 🤔 
<img width="866" alt="Screenshot 2024-01-18 at 11 31 30" src="https://github.com/mui/mui-x/assets/4941090/5fc04409-a5d4-4f5f-b442-443e5780a6ac">

Reply:
>  why other slots don't have it?

Why other slots don't have a class name?
Tbh I'm discovering the new infra bits by bits :laughing:

Reply:
Are they `styled` components?
I imagine that only `styled` components would get a `className`. 🤔

Reply:
> Are they `styled` components?

My advice from some time in this code is "don't over estimate its ability"

For classes, it relies on the classes type convention (cf [this code](https://github.com/mui/material-ui/blob/master/packages/api-docs-builder/utils/parseSlotsAndClasses.ts/#L97) ``const classesInterfaceName = `${componentName}Classes`;``) to find the typing of classes

Then it do some processing on it.


> Why not over commponents have classes?

Because only root has a type
```ts
export interface RichTreeViewClasses {
  /** Styles applied to the root element. */
  root: string;
}
```

Reply:
Thanks for the clarification. 👍

Reply:
From what I understand the classes are generated from the `RichTreeViewClasses` interface (for `RichTreeView`).
They don't have to be applied to elements in `styled`.

Reply:
I missed Alex answer :see_no_evil: